### PR TITLE
To-do-list App

### PR DESCRIPTION
**Pull request by ifeanyi305**
I added the CRUD method to my to-do list project.
The Add and Remove button is active
The added task can be updated 
The to-do list was built with HTML, js, CSS, and webpack
linters errors are fixed

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -2,48 +2,24 @@
...
 
-  const list = document.createElement('li');
-  list.setAttribute('class', 'list');
-  list.setAttribute('id', index);
-  ulLists.appendChild(list);
+// addBtn.addEventListener('click', (event) => {
+//   event.preventDefault();
+//   addList(todoTask.value, false, localGet.length + 1);
+//   displayList();
+// })
```
Comment (aimalamiri) [inline src/index.js]: It's highly recommended to remove the commented code from your project, please make sure to keep your repository as clean as possible.

File: src/modules/app.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import localGet from './localstorage.js';
+
+const todoContent = document.getElementById('listItem');
+const todoTask = document.getElementById('input');
+const displayList = () => {
+  todoContent.replaceChildren();
+  localGet().forEach((item, id) => {
+    todoContent.innerHTML += `
+        <div class="toDoItem">
+        <input class='item' id='check-${id}', "completed")' type='checkbox' ${item.Checked ? 'true' : 'false'} onclick='updateList(${id}, "completed")'>
```
Comment (aimalamiri) [inline src/modules/app.js]: Good job! but it has a small issue, when I click on the checkbox it does complete the task, but when I refresh the page the checkbox does not remain checked. Please make sure to `check` the checkbox for completed tasks.

File: src/modules/app.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const filteredArray = localGet().filter((item) => {
+    if (item.index !== id) {
+      return item;
+    }
+    return '';
+  });
+  localStorage.setItem('listStorage', JSON.stringify(filteredArray));
+
+  displayList();
+};
```
Comment (aimalamiri) [inline src/modules/app.js]: It works great, but it's part of the project requirement to update the index property for all of the tasks when you remove a task.

image

File: src/modules/app.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+window.updateList = (id) => {
+  const updateInput = document.querySelector(`#input-${id}`).value;
+  const updateArray = localGet().map((item) => {
+    if (item.index - 1 === id) {
+      item.description = updateInput;
+    }
```
Comment (aimalamiri) [inline src/modules/app.js:57]: Good job! but it's required to update the localStorage when you update the task description. This helps to persist the data. Please make sure to fix this.

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi @ifeanyi305, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights

- The app is designed very well and look fantastic ✔️ ⭐
- Good job for using the right GitHub Flow 💯
- Awesome file organization 🗃️
- Commits are descriptive and meaningful 👍
- Linters are properly set up and passing 🟢 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/app.js (hunk 2 above)
Comment (Kamba56) [reply src/modules/app.js]: Hi @aimalamiri, I think this functionality is for the next milestone (interactive list). This is the add remove milestone.

File: src/index.html [hunk 5]
```diff
@@ -11,14 +11,13 @@
...
-            </div>
-            <div class="listItem"></div>
-            <div class="ul-items"></div>
+                <button type="submit" id="added"><i class=" add">add</i></button>
+            </form>
+            <div id="listItem"></div>
             <div class="btn-con">
-                <button class="Completedbtn">Clear Completed Tasks</button>
+                <button id="Completedbtn">Clear Completed Tasks</button>
             </div>
```
Comment (Kamba56) [inline src/index.html:21]: - [Optional] Consider using an indentation of two spaces in all your documents. This is a convention and also [best practice.](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)

**Review by Kamba56 — APPROVED** [review]
Hi @ifeanyi305, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] All linters passed!👏👏👏
- [x] Great use of Github workflow!👏👏👏
- [x] Descriptive PR!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html (hunk 5 above)
Comment (ifeanyi305) [reply src/index.html:21]: Thank you

**Pull request merged**